contracts/chain-adapters/ForwarderBase.sol

```diff
@@ -98,8 +96,15 @@ abstract contract ForwarderBase is UUPSUpgradeable, ForwarderInterface {
      */
     function updateAdapter(uint256 _destinationChainId, address _l2Adapter) external onlyAdmin {
         if (_l2Adapter == address(0)) revert InvalidChainAdapter();
-        chainAdapters[_destinationChainId] = _l2Adapter;
-        emit ChainAdaptersUpdated(_destinationChainId, _l2Adapter);
+        _updateAdapter(_destinationChainId, _l2Adapter);
+    }
+
+    /**
+     * @notice Removes this contract's set adapter for the specified chain ID.
+     * @param _destinationChainId The chain ID of the target network.
+     */
+    function removeAdapter(uint256 _destinationChainId) external onlyAdmin {
```

Comment:
Do you think we should allow `updateAdapter` to be called with `address(0)` as the `_l2Adapter` to allow the admin to "remove" an adapter? I'm pretty indifferent, the tradeoff would be a smaller contract interface versus adding this intuitive function and preventing resetting adapter to address(0) accidentally

Reply:
I originally separated the two since I didn't want to run the risk of inadvertently setting the adapter to 0, but on second thought, we really won't run into this issue, since I'd think we would catch the error before executing the update transaction. I removed it https://github.com/across-protocol/contracts/pull/728/commits/8e4376964270bb50edfafa3d41becf02e9e81823

Reply:
yeah I don't think that's a risk since its an onlyAdmin function and we can assume the admin wouldn't do this